packages/authentication/docs/authentication-action.md

```diff
@@ -0,0 +1,26 @@
+### Auth action
+
+```ts
+import * as HttpErrors from 'http-errors';
+
+async action(request: Request): Promise<UserProfile | undefined> {
+    const authStrategy = await this.getAuthStrategy();
+    if (!authStrategy) {
+      // The invoked operation does not require authentication.
+      return undefined;
+    }
+
+    try {
+      const userProfile: UserProfile = await authStrategy.authenticate(request);
+      this.setCurrentUser(userProfile);
+      // a convenient return for the next request handlers
+      return userProfile;
+    } catch (err) {
+      // interpret the raw error code/msg here and throw the corresponding HTTP error
+      // convert it to http error
+      if (err.code == '401') {
+        throw new HttpErrors.Unauthorized(err.message);
+      }
```

Comment:
Shouldn't we throw for the `else` condition as well?

Reply:
good point 👍 The code in the .md file is pseudo code so I didn't refine too much for the error catch, will handle all the errors in story https://github.com/strongloop/loopback-next/issues/2467